src/FEValues/face_values.jl

```diff
@@ -186,6 +100,52 @@ Return the normal at the quadrature point `qp` for the active face of the
 """
 getnormal(fv::FaceValues, qp::Int) = fv.normals[qp]
 
+nfaces(fv::FaceValues) = length(fv.geo_values)
+
+function checkface(fv::FaceValues, face::Int)
+    0 < face <= nfaces(fv) || error("Face index out of range.")
+    return nothing
+end
+
+function reinit!(fv::FaceValues, x::AbstractVector{Vec{dim,T}}, face_nr::Int, cell=nothing) where {dim, T}
+    check_reinit_sdim_consistency(:FaceValues, shape_gradient_type(fv), eltype(x))
+    @boundscheck checkface(fv, face_nr)
+    n_geom_basefuncs = getngeobasefunctions(fv)
+    length(x) == n_geom_basefuncs || throw_incompatible_coord_length(length(x), n_geom_basefuncs)
+    
+    fv.current_face[] = face_nr
+
+    geo_values = get_geo_values(fv)
+    fun_values = get_fun_values(fv)
+    @inbounds for (q_point, w) in pairs(getweights(fv.qr, face_nr))
+        mapping = calculate_mapping(geo_values, q_point, x)
+        J = getjacobian(mapping)
+        weight_norm = weighted_normal(J, getrefshape(geo_values.ip), face_nr)
+        detJ = norm(weight_norm)
+        detJ > 0.0 || throw_detJ_not_pos(detJ)
```

Comment:
Can we even hit this case?

Reply:
I guess only in theory if `detJ==0`, but this check is the same as on master.
Should it be checked that `weighted_norm > 0`, or is this not required?

Reply:
This can happen if the ordering of the cells is reverted, which happens from time to time to new users when loading a mesh with Gmsh

Reply:
Then you get -0.5 instead of 0.5

Reply:
For the element, yes. But for the face? I thought the normal just flips around and there is no sane way to check if its the correct direction.

Reply:
I thought in 3D it's the same as for a quad in 2D

Reply:
Quad in 2D takes det(J), face in 3D (as seen in the line above) should be really just the norm of the weighted normal.